resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/NullFileSystem.java

```diff
@@ -26,26 +26,20 @@
  */
 public class NullFileSystem extends FileSystem {
 
-	/**
-	 * The singleton instance of this file system.
-	 */
-	private static IFileSystem instance;
-
 	/**
 	 * Returns the instance of this file system
 	 *
 	 * @return The instance of this file system.
 	 */
 	public static IFileSystem getInstance() {
-		return instance;
+		return EFS.getNullFileSystem();
```

Comment:
Nice trick!

Reply:
i don't like that this implementation unnecessarily looks up in a hashmap introducing unneeded computational complexity for an often used function.